OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseDatesPageFragment.kt

```diff
@@ -82,15 +84,9 @@ class CourseDatesPageFragment : OfflineSupportBaseFragment() {
         override fun onClick(link: String, blockId: String) {
             val component =
                 courseManager.getComponentByIdFromAppLevelCache(courseData.courseId, blockId)
+
             if (blockId.isNotEmpty() && component != null) {
-                val courseUnitDetailIntent = environment.router.getCourseUnitDetailIntent(
-                    requireActivity(),
-                    courseData,
-                    null,
-                    blockId,
-                    false
-                )
-                courseUnitDetailLauncher.launch(courseUnitDetailIntent)
+                showUnitDetailScreen(component)
```

Comment:
either revert this or move the analytics within the method. 
do the same with else condition.